lib/src/value/color.ts

```diff
@@ -3,371 +3,1112 @@
+            options.blue ?? NaN,
+          ],
+          alpha,
+        });
+        break;
+
+      case 'hsl': {
+        this.channel0Id = 'hue';
+        this.channel1Id = 'saturation';
+        this.channel2Id = 'lightness';
+        const hue = normalizeHue(options.hue ?? NaN);
+        const saturation = options.saturation ?? NaN;
+        let lightness = options.lightness ?? NaN;
+        lightness = assertClamped(lightness, 0, 100, 'lightness');
+        this.color = new Color({
+          spaceId: encodeSpaceForColorJs(space),
+          coords: [hue, saturation, lightness],
+          alpha,
+        });
+        break;
+      }
+
+      case 'hwb': {
+        this.channel0Id = 'hue';
+        this.channel1Id = 'whiteness';
+        this.channel2Id = 'blackness';
+        const hue = normalizeHue(options.hue ?? NaN);
+        const whiteness = options.whiteness ?? NaN;
+        const blackness = options.blackness ?? NaN;
+        this.color = new Color({
+          spaceId: encodeSpaceForColorJs(space),
+          coords: [hue, whiteness, blackness],
+          alpha,
+        });
+        break;
+      }
+
+      case 'lab':
+      case 'oklab': {
+        this.channel0Id = 'lightness';
+        this.channel1Id = 'a';
+        this.channel2Id = 'b';
+        let lightness = options.lightness ?? NaN;
+        const a = options.a ?? NaN;
+        const b = options.b ?? NaN;
+        const maxLightness = space === 'lab' ? 100 : 1;
+        lightness = assertClamped(lightness, 0, maxLightness, 'lightness');
+        this.color = new Color({
+          spaceId: encodeSpaceForColorJs(space),
+          coords: [lightness, a, b],
+          alpha,
+        });
+        break;
+      }
+
+      case 'lch':
+      case 'oklch': {
+        this.channel0Id = 'lightness';
+        this.channel1Id = 'chroma';
+        this.channel2Id = 'hue';
+        let lightness = options.lightness ?? NaN;
+        const chroma = options.chroma ?? NaN;
+        const hue = normalizeHue(options.hue ?? NaN);
+        const maxLightness = space === 'lch' ? 100 : 1;
+        lightness = assertClamped(lightness, 0, maxLightness, 'lightness');
+        this.color = new Color({
+          spaceId: encodeSpaceForColorJs(space),
+          coords: [lightness, chroma, hue],
+          alpha,
+        });
+        break;
+      }
+
+      case 'xyz':
+      case 'xyz-d65':
+      case 'xyz-d50':
+        this.channel0Id = 'x';
+        this.channel1Id = 'y';
+        this.channel2Id = 'z';
+        this.color = new Color({
+          spaceId: encodeSpaceForColorJs(space),
+          coords: [options.x ?? NaN, options.y ?? NaN, options.z ?? NaN],
+          alpha,
+        });
+        break;
     }
 
-    this.alphaInternal =
-      color.alpha === undefined
-        ? 1
-        : fuzzyAssertInRange(color.alpha, 0, 1, 'alpha');
+    // @TODO ColorJS doesn't seem to allow initial `alpha` to be missing?
+    if (this.alphaMissing) {
+      this.color.alpha = NaN;
+    }
```

Comment:
Why are we tracking `this.alphaMissing` explicitly if we can just set `color.alpha` to `NaN`?

Reply:
Removed in https://github.com/sass/embedded-host-node/pull/259/commits/18ab10ee84ce3b3321b8a8b98f52c7fb79650653